zerver/management/commands/delete_realm.py

```diff
@@ -28,6 +29,12 @@ def handle(self, *args: Any, **options: str) -> None:
 
         print(f"This realm has {user_count} users and {message_count} messages.\n")
 
+        if settings.CORPORATE_ENABLED:
+            from corporate.models import get_customer_by_realm
+
+            if get_customer_by_realm(realm):
```

Comment:
I think we want to check for `get_current_plan_by_realm` instead.  Having a customer object is necessary for having a paid plan but having a customer doesn't mean there is a paid plan. Realms which used to have a paid plan will return None for `get_current_plan_by_realm` but a non None value for `get_customer_by_realm`.

Reply:
I think we probably shouldn't permanently delete any realms that we've had a billing relationship with -- regardless of if they currently have one.  The bar for "no, you can't (or shouldn't) permanently delete this" should be quite low.

Reply:
Yup. Sounds good to me :)